src/shared/dlt_common.c

```diff
@@ -1815,35 +1828,115 @@ void dlt_print_with_attributes(bool state)
 }
 
 DltReturnValue dlt_log_init(int mode)
+{
+    return dlt_log_init_multiple_logfiles_support((DltLoggingMode)mode, false, 0, 0);
+}
+
+DltReturnValue dlt_log_init_multiple_logfiles_support(const DltLoggingMode mode, const bool enable_multiple_logfiles,
+                                            const int logging_file_size, const int logging_files_max_size)
 {
     if ((mode < DLT_LOG_TO_CONSOLE) || (mode > DLT_LOG_DROPPED)) {
-        dlt_user_printf("Wrong parameter for mode: %d\n", mode);
+        dlt_vlog(LOG_WARNING, "Wrong parameter for mode: %d\n", mode);
         return DLT_RETURN_WRONG_PARAMETER;
     }
 
     logging_mode = mode;
 
-    if (logging_mode == DLT_LOG_TO_FILE) {
-        /* internal logging to file */
-        logging_handle = fopen(logging_filename, "a");
+    if (logging_mode != DLT_LOG_TO_FILE) {
+        return DLT_RETURN_OK;
+    }
 
-        if (logging_handle == NULL) {
-            dlt_user_printf("Internal log file %s cannot be opened!\n", logging_filename);
-            return DLT_RETURN_ERROR;
+    if (enable_multiple_logfiles) {
+        dlt_user_printf("configure dlt logging using file limits\n");
```

Comment:
Sorry for let you waiting, but the amount of changes really take time to review. I have tested your changes and they work fine on my systems. Only thing I noticed is that above log message is printed twice. It is called from these two locations:
First call:
#0  dlt_log_init_multiple_logfiles_support (mode=DLT_LOG_TO_FILE, enable_multiple_logfiles=true, logging_file_size=200, logging_files_max_size=1000)
    at dlt-daemon/src/shared/dlt_common.c:1837
#1  0x000055555555d624 in main (argc=1, argv=0x7fffffffdb78) at /home/mmethner/src/dlt_size_limits/dlt-daemon/src/daemon/dlt-daemon.c:960


Second call:
#0  dlt_log_init_multiple_logfiles_support (mode=DLT_LOG_TO_FILE, enable_multiple_logfiles=true, logging_file_size=200, logging_files_max_size=1000)
    at dlt-daemon/src/shared/dlt_common.c:1837
#1  0x00005555555617c3 in dlt_daemon_local_init_p1 (daemon=<optimized out>, daemon_local=0x7fffffff7370, verbose=<optimized out>)
    at dlt-daemon/src/daemon/dlt-daemon.c:1198
#2  0x000055555555d792 in main (argc=<optimized out>, argv=<optimized out>)
    at dlt-daemon/src/daemon/dlt-daemon.c:1007
(gdb)

Reply:
Hi Michael,
thanks for the response :)

I use the already included code paths. As far as I understood the existing code this codeblock init the early log (e.g. for some startup logs) https://github.com/COVESA/dlt-daemon/blob/master/src/daemon/dlt-daemon.c#L944 and then reinit after forking https://github.com/COVESA/dlt-daemon/blob/master/src/daemon/dlt-daemon.c#L1171 . The log initializations are therefore executed twice as before (and create so the duplicated log line).

Is this pattern wrong? Is the logline really a problem? If yes, do you have a suggestion how I should restructure it?

Thanks in advanced

Reply:
Hello @danielweber2018 ,
i did not see any concrete issues (like memory leaks) or so due to the double call to this function.

As you correctly said, the behavior was there even before your patch, so I will continue with merging this patch and I have created an issue to track the double initialization (#452 ).

Thank you very much for your contribution and your long lasting support to get this patch merged :+1: